components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java

```diff
@@ -139,11 +141,36 @@ public boolean checkUsernameRecovery(String tenant) throws PreferenceRetrievalCl
      * @return returns true if  notification based password recovery enabled.
      * @throws PreferenceRetrievalClientException
      */
+    @Deprecated
     public boolean checkNotificationBasedPasswordRecovery(String tenant) throws PreferenceRetrievalClientException {
 
         return checkPreference(tenant, RECOVERY_CONNECTOR, NOTIFICATION_PASSWORD_RECOVERY_PROPERTY);
     }
 
+    /**
+     * Check email link based password recovery is enabled or not.
+     *
+     * @param tenant tenant domain name.
+     * @return returns true if  email link based password recovery enabled.
+     * @throws PreferenceRetrievalClientException
+     */
+    public boolean checkEmailLinkBasedPasswordRecovery(String tenant) throws PreferenceRetrievalClientException {
+
+        return checkPreference(tenant, RECOVERY_CONNECTOR, EMAIL_LINK_PASSWORD_RECOVERY_PROPERTY);
+    }
+
+    /**
+     * Check notification based password recovery is enabled or not.
+     *
+     * @param tenant tenant domain name.
```

Comment:
Shall we start the sentence with a capital letter?

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/5601/commits/2c29350aad74d55ec4a6243007ac968e35cc02f1